### PR TITLE
Ctrl+D Duplication

### LINKED ISSUE #1149: Add Ctrl+D for duplicating
Self explanatory

### PR DESCRIPTION
**Pull request by joeyballentine**
Closes #1149

### CODE REVIEW

File: src/renderer/contexts/GlobalNodeState.tsx [hunk 1]
```diff
@@ -1067,18 +1067,21 @@ export const GlobalProvider = memo(
...
                     return [...setSelected(nodes, false), ...newNodes];
                 });
 
                 changeEdges((edges) => {
                     const newEdge = copyEdges(
                         edges.filter((e) => {
-                            return nodesToCopy.has(e.target) && nodesToCopy.has(e.source);
+                            return (
+                                (nodesToCopy.has(e.target) && nodesToCopy.has(e.source)) ||
+                                edgeIds.includes(e.id)
```
Comment (RunDevelopment) [inline src/renderer/contexts/GlobalNodeState.tsx:1083]: Question: Users are able to select arbitrary edges. What will this do when the user selects a single edge and nothing else? What will the result of the user copying that be?

More general: How is copying edges supposed to behave?

Comment (joeyballentine) [reply src/renderer/contexts/GlobalNodeState.tsx:1083]: When i tested it, nothing happened (as expected) but i suppose it's possible it was still duplicating it. I didn't see any errors in the console though

Comment (RunDevelopment) [reply src/renderer/contexts/GlobalNodeState.tsx:1083]: Try this:
1. Create a chain with a lest one edge.
2. Select one edge and nothing else.
3. <kbd>Ctrl+D</kbd>
4. You will see that the edge you previously selected was deselected. Select it again by clicking on it.
5. Press <kbd>DEL</kbd>
6. You will see that the edge will be deselected.

What actually happened is that we duplicated the edge and then removed the duplicate again.

Comment (joeyballentine) [reply src/renderer/contexts/GlobalNodeState.tsx:1083]: Ah. I'll fix that then

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**